File: leetcode.com/python/852_Peak_Index_in_a_Mountain_Array.py
class Solution(object):
    def peakIndexInMountainArray(self, A):
        """
        :type A: List[int]
        :rtype: int
        """
        leftIdx, rightIdx = 0, len(A) - 1
        while leftIdx < rightIdx:
            midIdx = leftIdx + (rightIdx - leftIdx) // 2
            if A[midIdx] < A[midIdx + 1]:
                leftIdx = midIdx + 1
            else:
                rightIdx = midIdx
        return leftIdx

File: leetcode.com/swift/DS_ALGO_PRAC.playground/Sources/swift/852_Peak_Index_in_a_Mountain_Array.swift
import Foundation

class Solution {
    func peakIndexInMountainArray(_ A: [Int]) -> Int {
        var leftIdx = 0, rightIdx = A.count - 1
        while leftIdx < rightIdx {
            var midIdx = leftIdx + Int((rightIdx - leftIdx) / 2)
            if A[midIdx] < A[midIdx + 1] {
                leftIdx = midIdx + 1
            } else {
                rightIdx = midIdx
            }
        }
        return leftIdx
    }
}
